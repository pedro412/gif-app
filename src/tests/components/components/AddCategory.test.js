import React from 'react';
import '@testing-library/jest-dom';
const { shallow } = require('enzyme');
const { default: AddCategory } = require('../../../components/AddCategory');

describe('Testing <AddCategory />', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('it should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it should change the input box', () => {
    const input = wrapper.find('input');
    const value = 'asd';
    input.simulate('change', { target: { value } });
  });

  test('should not send submit if string length is < 2', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call setCategories and clean input', () => {
    const value = 'asd';
    const input = wrapper.find('input');

    input.simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  });
});
