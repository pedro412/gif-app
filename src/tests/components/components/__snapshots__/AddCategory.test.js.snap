// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AddCategory /> it should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id="value"
    name="value"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
